{
  "modified": "0001-01-01T00:00:00Z",
  "published": "2019-01-03T00:00:00Z",
  "schema_version": "1.5.0",
  "id": "PSF-2019-17",
  "summary": "http.cookiejar: Incorrect validation of path",
  "details": "Cookies of ``example.com`` with ``path=/any`` were sent to\n``example.com/anybad/`` while using a cookiejar with\n`http.cookiejar.DefaultCookiePolicy` policy. The code did not check for\nthe first non-matching character in prefix match to be a slash.\n",
  "affected": [
    {
      "ranges": [
        {
          "type": "GIT",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "0e1f1f01058bd4a9b98cfe443214adecc019a38c"
            },
            {
              "fixed": "382981b25092b5e9285f1e4894142af1e8f2ca86"
            },
            {
              "fixed": "5565b1db6f37f244890369e0d68a3e906aca28b9"
            },
            {
              "fixed": "97c7d78fda49e03fc773c171ce0c736d02bb73f5"
            },
            {
              "fixed": "e260f092cd0d8975c777e73ca6fb549d59b5d452"
            },
            {
              "fixed": "ee15aa2b8501718cb77e339381f72409a416f801"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [
    {
      "type": "REPORT",
      "url": "https://bugs.python.org/issue35647"
    }
  ],
  "database_specific": {
    "cwe_ids": []
  }
}
